src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -978,6 +977,50 @@ public void UpdatePatchDetails(Patch patch)
         UpdateReportTabPatchName(TranslationServer.Translate(patch.Name));
     }
 
+    /// <summary>
+    ///   Updates the existing part selection buttons with the correct properties based on their Node names.
+    /// </summary>
+    public void UpdateMicrobePartSelections()
+    {
+        foreach (var organelle in SimulationParameters.Instance.GetAllOrganelles())
+        {
+            foreach (var partSelection in placeablePartSelectionElements)
```

Comment:
It would be nice to remove this O(n^2) operation here. Instead as special classes are used anyway the part selection should hold a reference to the Organelle it is for after looking it up once.